File: LC2.java
/**
 * Definition for singly-linked list. public class ListNode { int val; ListNode
 * next; ListNode(int x) { val = x; } }
 */
class Solution {
	public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
		int in = 0;
		ListNode rootListNode = l1;
		while (l1 != null || l2 != null) {
			// System.out.println(in);
			// if (l1 != null && l2 != null) {
			int value = l1.val + l2.val + in;
			in = value / 10;
			l1.val = value % 10;
			if (l1.next == null && l2.next == null)
				break;
			if (l1.next == null)
				l1.next = new ListNode(0);
			if (l2.next == null)
				l2.next = new ListNode(0);
			l1 = l1.next;
			l2 = l2.next;
		}
		if (in != 0)

		{
			// System.out.println(rootListNode.next);
			l1.next = new ListNode(in);
			// System.out.println(rootListNode.next);
		}
// System.out.println(in);
		return rootListNode;
	}
}

File: LC2.go
package main

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	in := 0
	yummy := &ListNode{-1, nil}
	prev := yummy
	for l1 != nil || l2 != nil {
		l := 0
		r := 0
		if l1 != nil {
			l = l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			r = l2.Val
			l2 = l2.Next
		}
		sum := l + r + in
		in = sum / 10
		prev.Next = &ListNode{sum % 10, nil}
		prev = prev.Next
	}
	if in != 0 {
		prev.Next = &ListNode{in, nil}
	}
	return yummy.Next
}
